blueman/gui/manager/ManagerDeviceList.py

```diff
@@ -450,15 +466,17 @@ def _update_power_levels(self, tree_iter: Gtk.TreeIter, device: Device, cinfo: c
 
         bars = {}
 
-        if device["ServicesResolved"] and any(ServiceUUID(uuid).short_uuid == BATTERY_SERVICE_SVCLASS_ID
-                                              for uuid in device["UUIDs"]):
-            bars["battery"] = Battery(obj_path=device.get_object_path())["Percentage"]
+        obj_path = device.get_object_path()
+        if obj_path in self._batteries:
+            bars["battery"] = self._batteries[obj_path]["Percentage"]
 
         # cinfo init may fail for bluetooth devices version 4 and up
         # FIXME Workaround is horrible and we should show something better
         if cinfo.failed:
             if not bars:
                 bars = {"rssi": 100.0, "tpl": 100.0, "lq": 100.0}
+            else:
+                bars.update({"rssi": 100.0, "tpl": 100.0, "lq": 100.0})
```

Comment:
Could be just the `else` branch, no?

Reply:
we don't even need to branch, we can use dict.update in both cases. I'll fix that.

Reply:
That's what I tried to express. :sweat_smile: